src/coreclr/vm/prestub.cpp

```diff
@@ -1067,7 +1054,436 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+    //  * Constructor access - examine the return type
+    //  * Instance member access - examine type of first parameter
+    //  * Static member access - examine type of first parameter
+    TypeHandle retType;
+    TypeHandle firstArgType;
+    retType = context.DeclarationSig.GetRetTypeHandleThrowing();
+    UINT argCount = context.DeclarationSig.NumFixedArgs();
+    if (argCount > 0)
+    {
+        context.DeclarationSig.NextArg();
+        firstArgType = context.DeclarationSig.GetLastTypeHandleThrowing();
+    }
+
+    // Using the kind type, perform the following:
+    //  1) Validate the basic type information from the signature.
+    //  2) Resolve the name to the appropriate member.
+    //  3) Generate the IL for the accessor.
+    switch (context.Kind)
+    {
+    case UnsafeAccessorKind::Constructor:
+        // A return type is required for a constructor, otherwise
+        // we don't know the type to construct.
+        // The name is defined by the runtime and should be empty.
+        if (retType.IsNull() || !name.IsEmpty())
+        {
+            ThrowHR(COR_E_BADIMAGEFORMAT);
+        }
+
+        context.TargetType = retType;
+        if (!TrySetTargetMethodCtor(context)
+            || !TryGenerateAccessor(context, resolver, methodILDecoder))
+        {
+            ThrowHR(COR_E_MISSINGMETHOD);
+        }
+        break;
+
+    case UnsafeAccessorKind::Method:
+    case UnsafeAccessorKind::StaticMethod:
+        // Method access requires a target type.
+        if (firstArgType.IsNull())
+            ThrowHR(COR_E_BADIMAGEFORMAT);
+
+        context.TargetType = firstArgType;
+        context.IsTargetStatic = kind == UnsafeAccessorKind::StaticMethod;
+        if (!TrySetTargetMethod(context, name.GetUTF8())
+            || !TryGenerateAccessor(context, resolver, methodILDecoder))
+        {
+            ThrowHR(COR_E_MISSINGMETHOD);
+        }
+        break;
+
+    case UnsafeAccessorKind::Field:
+    case UnsafeAccessorKind::StaticField:
+        // Field access requires a single argument for target type and a return type.
+        if (argCount != 1 || firstArgType.IsNull() || retType.IsNull())
+        {
+            ThrowHR(COR_E_BADIMAGEFORMAT);
+        }
+
+        // The return type must be byref.
+        // If the non-static field access is for a
+        // value type, the instance must be byref.
+        if (!retType.IsByRef()
+            || (kind == UnsafeAccessorKind::Field
+                && firstArgType.IsValueType()
+                && !firstArgType.IsByRef()))
+        {
+            ThrowHR(COR_E_BADIMAGEFORMAT);
+        }
+
+        context.TargetType = firstArgType;
+        context.IsTargetStatic = kind == UnsafeAccessorKind::StaticField;
+        if (!TrySetTargetField(context, name.GetUTF8(), retType.GetTypeParam())
+            || !TryGenerateAccessor(context, resolver, methodILDecoder))
+        {
+            ThrowHR(COR_E_MISSINGFIELD);
```

Comment:
```suggestion
            MemberLoader::ThrowMissingFieldException(firstArgType.GetMethodTable(), name.GetUTF8());
```
Better error message?

Reply:
Dtto for methods.